pkgs/build-support/rust/default.nix

```diff
@@ -26,10 +29,13 @@
 , cargoBuildFlags ? []
 , buildType ? "release"
 , meta ? {}
-, target ? null
+, target ? rust.toRustTarget stdenv.hostPlatform
 , cargoVendorDir ? null
 , checkType ? buildType
 
+# Only matters if `target` is a JSON file. Toggles whether a custom sysroot is created.
+, dontAddSysroot ? false
```

Comment:
This needs documentation in `nixpkgs/doc/languages-frameworks/rust.section.md`
Also it should explain what the implication of custom sysroot is and when one 
wants to use it ideally with an example.